code/modules/fishing/fishing_minigame.dm

```diff
@@ -359,16 +360,32 @@ GLOBAL_LIST_EMPTY(fishing_challenges_by_user)
 		user.mind.adjust_experience(/datum/skill/fishing, round(seconds_spent * FISHING_SKILL_EXP_PER_SECOND * experience_multiplier))
 		if(user.mind.get_skill_level(/datum/skill/fishing) >= SKILL_LEVEL_LEGENDARY)
 			user.client?.give_award(/datum/award/achievement/skill/legendary_fisher, user)
-	if(win)
-		if(reward_path != FISHING_DUD)
-			playsound(location, 'sound/effects/bigsplash.ogg', 100)
-		if(ispath(reward_path, /obj/item/fish) || isfish(reward_path))
-			var/obj/item/fish/fish_reward = reward_path
-			var/obj/item/fish/redirect_path = initial(fish_reward.fish_id_redirect_path)
-			var/fish_id = ispath(redirect_path, /obj/item/fish) ? initial(redirect_path.fish_id) : initial(fish_reward.fish_id)
-			if(fish_id)
-				user.client?.give_award(/datum/award/score/progress/fish, user, fish_id)
-	SEND_SIGNAL(user, COMSIG_MOB_COMPLETE_FISHING, src, win)
+
+	if(!win)
+		SEND_SIGNAL(user, COMSIG_MOB_COMPLETE_FISHING, src, FALSE)
+		if(!QDELETED(src))
+			qdel(src)
+		return
+
+	if(reward_path != FISHING_DUD)
+		playsound(location, 'sound/effects/bigsplash.ogg', 100)
+
+	var/valid_achievement_catch = FALSE
+	if (ispath(reward_path, /obj/item/fish))
+		valid_achievement_catch = TRUE
+	else if (isfish(reward_path))
+		var/obj/item/fish/fishy_individual = reward_path
+		if (!HAS_TRAIT(fishy_individual, TRAIT_NO_FISHING_ACHIEVEMENT))
```

Comment:
```suggestion
		if (!HAS_TRAIT(fishy_individual, TRAIT_NO_FISHING_ACHIEVEMENT) || fishy_individual.status == FISH_DEAD)
```
I kinda doubt this will be harsh, but I'll leave it as a suggestion rather than a request.

Reply:
I got the logic backwards so I took a second to fix it so you didn't have to.